Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -3663,9 +3663,9 @@ LayoutUnit RenderBox::availableLogicalHeightUsing(const Length& h, AvailableLogi
     if (std::optional<LayoutUnit> heightIncludingScrollbar = computeContentAndScrollbarLogicalHeightUsing(MainOrPreferredSize, h, std::nullopt))
         return std::max<LayoutUnit>(0, adjustContentBoxLogicalHeightForBoxSizing(heightIncludingScrollbar) - scrollbarLogicalHeight());
 
-    // FIXME: Check logicalTop/logicalBottom here to correctly handle vertical writing-mode.
-    // https://bugs.webkit.org/show_bug.cgi?id=46500
-    if (is<RenderBlock>(*this) && isOutOfFlowPositioned() && style().height().isAuto() && !(style().top().isAuto() || style().bottom().isAuto())) {
+    // Height of absolutely positioned, non-replaced elements section 5.3 rule 5
```

Comment:
This comment is wrong. This actually fixes a couple of tests which test sizing of replaced elements - a different part of the spec entirely.

https://wpt.fyi/results/css/css-sizing/intrinsic-percent-replaced-dynamic-008.html?label=experimental&label=master&aligned
https://wpt.fyi/results/css/css-sizing/intrinsic-percent-replaced-008.html?label=experimental&label=master&aligned

Reply:
Spoke with Alan about my confusion. The replaced elements are children of this block. It's not the replaced element that is being positioned absolutely, it's the containing block of the replaced element.

The writing mode of the block doesn't have an effect on the block itself, but rather on the children. So checking for logicalTop and logicalBottom in the cases of these tests is what's causing the containing block and therefore the canvas to be sized appropriately.

Reply:
In other words, the canvas is asking its containing block for how much height it can take up and so the containing block needs to return its answer in terms of the writing mode.